keras/src/visualization/draw_bounding_boxes.py

```diff
@@ -0,0 +1,177 @@
+        TypeError: If `bounding_boxes` is not a dictionary.
+        ImportError: If `cv2` (OpenCV) is not installed.
+    """
+
+    if cv2 is None:
+        raise ImportError(
+            "The `draw_bounding_boxes` function requires the `cv2` package "
+            " (OpenCV). Please install it with `pip install opencv-python`."
+        )
+
+    class_mapping = class_mapping or {}
+    text_thickness = (
+        text_thickness or line_thickness
+    )  # Default text_thickness if not provided.
+    data_format = data_format or backend.image_data_format()
+    images_shape = ops.shape(images)
+    if len(images_shape) != 4:
+        raise ValueError(
+            "`images` must be batched 4D tensor. "
+            f"Received: images.shape={images_shape}"
+        )
+    if not isinstance(bounding_boxes, dict):
+        raise TypeError(
+            "`bounding_boxes` should be a dict. "
+            f"Received: bounding_boxes={bounding_boxes} of type "
+            f"{type(bounding_boxes)}"
+        )
+    if "boxes" not in bounding_boxes or "labels" not in bounding_boxes:
+        raise ValueError(
+            "`bounding_boxes` should be a dict containing 'boxes' and "
+            f"'labels' keys. Received: bounding_boxes={bounding_boxes}"
+        )
+    if data_format == "channels_last":
+        h_axis = -3
+        w_axis = -2
+    else:
+        h_axis = -2
+        w_axis = -1
+    height = images_shape[h_axis]
+    width = images_shape[w_axis]
+    bounding_boxes = bounding_boxes.copy()
+    bounding_boxes = convert_format(
+        bounding_boxes, bounding_box_format, "xyxy", height, width
+    )
+
+    # To numpy array
+    images = ops.convert_to_numpy(images).astype("uint8")
+    boxes = ops.convert_to_numpy(bounding_boxes["boxes"])
+    labels = ops.convert_to_numpy(bounding_boxes["labels"])
+    if "confidences" in bounding_boxes:
+        confidences = ops.convert_to_numpy(bounding_boxes["confidences"])
+    else:
+        confidences = None
+
+    result = []
+    batch_size = images.shape[0]
+    for i in range(batch_size):
+        _image = images[i]
+        _box = boxes[i]
+        _class = labels[i]
+        for box_i in range(_box.shape[0]):
+            x1, y1, x2, y2 = _box[box_i].astype("int32")
+            c = _class[box_i].astype("int32")
+            if c == -1:
+                continue
+            x1, y1, x2, y2 = int(x1), int(y1), int(x2), int(y2)
+            c = int(c)
+            # Draw bounding box
+            cv2.rectangle(_image, (x1, y1), (x2, y2), color, line_thickness)
+
+            if c in class_mapping:
+                label = class_mapping[c]
+                if confidences is not None:
+                    conf = confidences[i][box_i]
+                    label = f"{label} | {conf:.2f}"
+
+                font_x1, font_y1 = _find_text_location(
+                    x1, y1, font_scale, text_thickness
+                )
+                cv2.putText(
+                    _image,
```

Comment:
Please use keyword arguments for better readability

Reply:
Done